have docs explain why we import guard (#80569)
ansible/ansible: docs/docsite/rst/dev_guide/testing/sanity/import.rst

Handling import errors
----------------------

> Ansible executes across many hosts and can uses different interpreters (normally python) that can even have multiple versions installed in parallel at the same time.

Reviewer: Since the import tests only apply to Python, I don't think we need to mention non-Python interpreters here.

```suggestion
Ansible executes across many hosts and can use multiple Python interpreters at the same time, which may even have different versions.
```

> To ensure users get an actionable and easy to understand error we try to ensure any non core libraries that a module/plugin are guarded in their imports to avoid a traceback,

Reviewer: ```suggestion
To ensure users get an actionable and easy to understand error we try to ensure any non-core imports in modules/plugins are guarded to avoid a traceback,
```

which most users won't be able to understand, much less use to solve the issue.

> Another reason we do this is to allow us to import the code for inspection. This is how we can easily test, document, configure, etc based on the code without having to install

Reviewer 2: ```suggestion
Another reason Ansible does this is to import the code for inspection. This allows Ansible to easily test, document, configure, etc based on the code without having to install
```

> Another reason we do this is to allow us to import the code for inspection, this is how we can easily test, document, configure, etc based on the code without having to install

Reviewer: ```suggestion
Another reason we do this is to allow us to import the code for inspection. This is how we can easily test, document, configure, etc based on the code without having to install
```

> any and all requirements everywhere, specially when that is not the context you might execute the code in.

Reviewer 2: ```suggestion
any and all requirements everywhere, especially when that is not the context in which you execute the code.
```

> The code below shows examples on how to avoid errors on import and then use the provided ``missing_required_lib`` to ensure the user knows which LIBRARY is missing,

Reviewer: ```suggestion
The code below shows examples of how to avoid errors on import and then use the provided ``missing_required_lib`` to ensure the user knows which LIBRARY is missing,
```

> on which HOST it is missing on and the specific INTERPRETER that requires it.

Reviewer 2: ```suggestion
on which HOST it is missing and the specific INTERPRETER that requires it.
```

   else:
       ANOTHER_LIBRARY_IMPORT_ERROR = None

> Then in the plugin code, for example in ``run`` method of the plugin (some plugins don't have a 'run' and will require it in the ``__init__`` method instead):

Reviewer: ```suggestion
Then in the plugin code, for example in the ``run`` method of the plugin (some plugins don't have a ``run`` method and will require it in the ``__init__`` method instead):
```
